### ISSUE TITLE
Mod centre should not use nested interactive controls

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#14100]
**Describe the bug**

Nested interactive controls (e.g. a link within a button) is not valid HTML, and causes content to be inaccessible to screen reader users.

In the mod center (`/mod`) each article in the list is a button with a nested link:

![screenshot showing the list of article in the mod center, with the axe devtools report open to the right. The error noted above is highlighted, as well as the link and the button](https://user-images.githubusercontent.com/20773163/123776582-53be3d00-d8c7-11eb-89d2-c1476e32ee47.png)

The buttons themselves are not very accessible - they don't describe to a screen reader user their function, or provide any feedback when pressed (a screen reader user would not be informed expandable content had appeared).

**To Reproduce**

- Visit `/mod` as a moderator
- Using a screen reader, press tab so that the first article button is highlighted. Notice the whole row content is announced as the button name
- Click the button by pressing enter
- Notice no feedback is provided to you
- Press tab again to focus the link of the article name within the button
- Notice nothing is announced 

**Expected behavior**

I think this section needs some refactoring. I would suggest making each article in the list a summary/details panel which has expand/collapse behaviour baked in to the HTML. The link to the article could then be added as the first item in the details panel before the remaining content.

**Additional context**

This issue was flagged when `axe-core` updated and our frontend tests for `app/javascript/modCenter/singleArticle/__tests__/singleArticle.test.jsx` started failing. When we address this issue, we need to update that test again to remove workarounds for the axe failure.

**Comment by cmgorton** [comment]
@aitchiss would it be ok if I add the MLH fellow label to this issue for them to potentially work on it?

**Comment by aitchiss** [comment]
Sounds good to me @cmgorton!

**Comment by aayush-05** [comment]
Can I pick this up?

**Comment by aayush-05** [comment]
@aitchiss do we want the button (for opening the post in a separate window) inside this iframe (refer image) or add a new row at the top of the iframes and add it there?
![image](https://user-images.githubusercontent.com/47032027/126356168-94900cbd-157a-49f2-9b51-597f5cd9511e.png)

**Comment by aitchiss** [comment]
> @aitchiss do we want the button (for opening the post in a separate window) inside this iframe (refer image) or add a new row at the top of the iframes and add it there?

I think it would work best as a new row before the iframe - what do you think?

**Comment by aayush-05** [comment]
Sounds good to me; let me do an initial version and then we can discuss on any change requirements
